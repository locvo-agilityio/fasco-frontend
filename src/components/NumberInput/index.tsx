'use client';

import { useEffect, useState } from 'react';
import { MinusIcon, PlusIcon } from 'lucide-react';

// Components
import { Button } from '@/components/common';

type NumberInputProps = {
  title?: string;
  min?: number;
  max?: number;
  step?: number;
  onChange?: (value: number) => void;
};

const NumberInput = ({
  title = '',
  min = 1,
  max = 100,
  step = 1,
  onChange,
}: NumberInputProps) => {
  const [value, setValue] = useState<number>(min);

  const handleDecrease = () => {
    const newValue = value - step;

    setValue(newValue);
    onChange?.(newValue);
  };

  const handleIncrease = () => {
    const newValue = value + step;

    setValue(newValue);
    onChange?.(newValue);
  };

  useEffect(() => {
    setValue(min);
  }, [min]);

  return (
    <div className="flex flex-col gap-2 w-fit">
      {title && <p className="font-volkhov text-base">{title}</p>}
      <div className="flex w-32 h-12 items-center gap-2 border border-gray-300 rounded-base">
        <Button
          title="Decrease"
          onPress={handleDecrease}
          isDisabled={value === 1}
          className="w-11 h-12 min-w-11 px-0 py-0 bg-transparent hover:bg-gray-300 rounded-none border-none"
        >
          <MinusIcon size={9} className="text-black" />
        </Button>
        <span className="text-base font-normal font-jost w-10 text-center">
          {value}
        </span>
        <Button
          title="Increase"
          onPress={handleIncrease}
          isDisabled={value >= max}
          className="w-11 h-12 min-w-11 bg-transparent hover:bg-gray-300 rounded-none border-none"
        >
          <PlusIcon size={9} className="text-black" />
        </Button>
      </div>
    </div>
  );
};

export default NumberInput;
