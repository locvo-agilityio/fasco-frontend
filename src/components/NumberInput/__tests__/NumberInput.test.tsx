import { fireEvent, render, waitFor } from '@testing-library/react';

// Components
import NumberInput from '..';

const mockOnChange = jest.fn();

describe('NumberInput', () => {
  it('should render correctly', () => {
    const { container } = render(<NumberInput />);

    expect(container).toMatchSnapshot();
  });

  it('handle click button', async () => {
    const { getByTitle } = render(<NumberInput onChange={mockOnChange} />);

    const decreaseButton = getByTitle('Decrease');
    const increaseButton = getByTitle('Increase');

    await waitFor(() => {
      fireEvent.click(increaseButton);
      fireEvent.click(increaseButton);
      fireEvent.click(increaseButton);
      fireEvent.click(decreaseButton);
    });

    await waitFor(() => {
      expect(decreaseButton).toBeInTheDocument();
      expect(increaseButton).toBeInTheDocument();
      expect(mockOnChange).toHaveBeenCalled();
    });
  });
});
