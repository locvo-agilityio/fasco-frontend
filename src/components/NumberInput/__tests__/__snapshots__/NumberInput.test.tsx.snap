// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput should render correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-2 w-fit"
  >
    <div
      class="flex w-32 h-12 items-center gap-2 border border-gray-300 rounded-base"
    >
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none border-default data-[hover=true]:opacity-hover border-1 text-foreground-100 disabled:opacity-50 text-base rounded-base data-[pressed=true]:scale-[1] font-poppins w-11 h-12 min-w-11 px-0 py-0 bg-transparent hover:bg-gray-300 rounded-none border-none"
        data-disabled="true"
        disabled=""
        title="Decrease"
        type="button"
      >
        <svg
          class="lucide lucide-minus text-black"
          fill="none"
          height="9"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="9"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
        </svg>
      </button>
      <span
        class="text-base font-normal font-jost w-10 text-center"
      >
        1
      </span>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none border-default data-[hover=true]:opacity-hover border-1 text-foreground-100 disabled:opacity-50 py-5 text-base rounded-base data-[pressed=true]:scale-[1] font-poppins w-11 h-12 min-w-11 bg-transparent hover:bg-gray-300 rounded-none border-none"
        title="Increase"
        type="button"
      >
        <svg
          class="lucide lucide-plus text-black"
          fill="none"
          height="9"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="9"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="M12 5v14"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
